static/js/components/ModifyInstrument.jsx

```diff
@@ -0,0 +1,281 @@
+  const filters = [...enum_types.ALLOWED_BANDPASSES];
+
+  if (telescopeList.length === 0 || instrumentList.length === 0) {
+    return (
+      <div>
+        <CircularProgress color="secondary" />
+      </div>
+    );
+  }
+
+  const telLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  telescopeList?.forEach((tel) => {
+    telLookUp[tel.id] = tel;
+  });
+
+  const instLookUp = {};
+  // eslint-disable-next-line no-unused-expressions
+  instrumentList?.forEach((instrumentObj) => {
+    instLookUp[instrumentObj.id] = instrumentObj;
+  });
+
+  const handleSelectedInstrumentChange = (e) => {
+    setSelectedInstrumentId(e.target.value);
+  };
+
+  function validate(formData, errors) {
+    const oldFilters = [];
+    instLookUp[selectedInstrumentId].filters?.forEach((filter) => {
+      oldFilters.push(filter);
+    });
+    const oldFiltersUnique = [...new Set(oldFilters)];
+
+    const newFilters = [];
+    formData.filters?.forEach((filter) => {
+      newFilters.push(filter);
+    });
+    const newFiltersUnique = [...new Set(newFilters)];
+
+    const result = oldFiltersUnique.every((val) =>
+      newFiltersUnique.includes(val)
+    );
+    if (errors && formData.filters && !result) {
+      errors.filters.addError("New filter list must contain old filter list.");
+    }
+    if (errors && formData.api_classname && formData.api_classname.length > 1) {
+      errors.api_classname.addError("Must only choose one API class.");
+    }
+    if (
+      errors &&
+      formData.api_classname_obsplan &&
+      formData.api_classname_obsplan.length > 1
+    ) {
+      errors.api_classname_obsplan.addError("Must only choose one API class.");
+    }
+    return errors;
+  }
+
+  const instrumentFormSchema = {
+    type: "object",
+    properties: {
+      filters: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: filters,
+        },
+        uniqueItems: true,
+        title: "Filter list",
+      },
+      api_classname: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: api_classnames,
+        },
+        uniqueItems: true,
+        title: "API Classname",
+      },
+      api_classname_obsplan: {
+        type: "array",
+        items: {
+          type: "string",
+          enum: api_classnames,
+        },
+        uniqueItems: true,
+        title: "API Observation Plan Classname",
+      },
+      field_data: {
+        type: "string",
+        format: "data-url",
+        title: "Field data file",
+        description: "Field data file",
+      },
+      field_region: {
+        type: "string",
+        format: "data-url",
+        title: "Field region file",
+        description: "Field region file",
+      },
+    },
+  };
+
+  return (
+    <div className={classes.container}>
+      <InputLabel id="instrumentSelectLabel">Instrument</InputLabel>
+      <Select
+        inputProps={{ MenuProps: { disableScrollLock: true } }}
```

Comment:
I'm curious: what does `disableScrollLock` do?

Reply:
i copied this from the others... ;) something about menu location relative to the parent menu.